libmamba/src/core/problems_graph_creator.cpp

```diff
@@ -0,0 +1,215 @@
+                auto target_node = MNode(NodeInfo::ProblematicPackage(problem.dep().value()),
+                                         from(problem.type));
+                auto edge = MEdge(EdgeInfo::Require(DependencyInfo(problem.dep().value())));
+                add_edge(problem.source_id, source_node, problem.target_id, target_node, edge);
+                break;
+            }
+            case SOLVER_RULE_PKG_CONFLICTS:
+            case SOLVER_RULE_PKG_SAME_NAME:
+            {
+                if (!has_values(problem, { problem.source(), problem.target() }))
+                {
+                    break;
+                }
+                auto source_node = MNode(NodeInfo::ResolvedPackage(problem.source().value()),
+                                         from(problem.type));
+                auto target_node = MNode(NodeInfo::ResolvedPackage(problem.target().value()),
+                                         from(problem.type));
+                add_conflicts(problem.source_id, source_node, problem.target_id, target_node);
+                break;
+            }
+            case SOLVER_RULE_UPDATE:
+            {
+                if (!has_values(problem, { problem.source() }))
+                {
+                    break;
+                }
+                LOG_WARNING << "Update rules are unsupported for now" << std::endl;
+                return;
+            }
+            case SOLVER_RULE_BEST:
+            case SOLVER_RULE_BLACK:
+            case SOLVER_RULE_INFARCH:
+            case SOLVER_RULE_PKG_NOT_INSTALLABLE:
+            case SOLVER_RULE_STRICT_REPO_PRIORITY:
+            {
+                if (!has_values(problem, { problem.source() }))
+                {
+                    break;
+                }
+                get_update_or_create(
+                    problem.source_id,
+                    MNode(NodeInfo::ResolvedPackage(problem.source().value()), from(problem.type)));
+                break;
+            }
+            case SOLVER_RULE_JOB_PROVIDED_BY_SYSTEM:
+            {
+                LOG_WARNING << "Problem type not implemented: " << problem.type;
+                break;
+            }
+            case SOLVER_RULE_CHOICE:
+            case SOLVER_RULE_FEATURE:
+            case SOLVER_RULE_JOB_UNSUPPORTED:
+            case SOLVER_RULE_LEARNT:
+            case SOLVER_RULE_PKG_RECOMMENDS:
+            case SOLVER_RULE_RECOMMENDS:
+            case SOLVER_RULE_UNKNOWN:
+            {
+                LOG_WARNING << "Problem type not implemented: " << problem.type;
+                break;
+            }
+            case SOLVER_RULE_PKG_SELF_CONFLICT:
+            case SOLVER_RULE_PKG_OBSOLETES:
+            case SOLVER_RULE_PKG_IMPLICIT_OBSOLETES:
+            case SOLVER_RULE_PKG_INSTALLED_OBSOLETES:
+            case SOLVER_RULE_YUMOBS:
+            case SOLVER_RULE_DISTUPGRADE:
+                LOG_WARNING << "Problem type not implemented: " << problem.type;
```

Comment:
Would that be similar?
```suggestion
            default:
                LOG_WARNING << "Problem type not implemented: " << problem.type;
```

Reply:
Currently, yes, but I think we might want to implement the first 2 blocks of cases, the third one i am not sure where it is used though, but we can merge them for now.